src/analytics/AnalyticsManager.ts

```diff
@@ -0,0 +1,497 @@
+ *
+ * OPT-OUT:
+ * To completely disable analytics:
+ * ```typescript
+ * const sdk = new NexusSDK({ analytics: { enabled: false } });
+ * ```
+ *
+ * PRIVACY CONTROLS:
+ * ```typescript
+ * const sdk = new NexusSDK({
+ *   analytics: {
+ *     privacy: {
+ *       anonymizeWallets: true,  // Hash wallet addresses with SHA-256
+ *       anonymizeAmounts: true,  // Exclude transaction amounts from events
+ *     }
+ *   }
+ * });
+ * ```
+ *
+ * CUSTOM ANALYTICS:
+ * You can use your own PostHog instance:
+ * ```typescript
+ * const sdk = new NexusSDK({
+ *   analytics: {
+ *     posthogApiKey: 'your-key',
+ *     posthogApiHost: 'https://your-posthog.com'
+ *   }
+ * });
+ * ```
+ */
+import { AnalyticsProvider } from './providers/AnalyticsProvider';
+import { PostHogProvider } from './providers/PostHogProvider';
+import { NoOpProvider } from './providers/NoOpProvider';
+import { SessionManager } from './session';
+import { PerformanceTracker } from './performance';
+import { NexusAnalyticsEvents } from './events';
+import { AnalyticsConfig, BaseEventProperties } from './types';
+import { extractErrorCode } from './utils';
+
+/**
+ * Default PostHog API Key - INTENTIONALLY PUBLIC
+ *
+ * This key is shared across all Nexus SDK users for Avail's internal telemetry.
+ * It is NOT a secret and is safe to commit to source code.
+ *
+ * Purpose:
+ * - Collect anonymous SDK usage metrics
+ * - Track errors and performance issues
+ * - Improve SDK quality for all users
+ *
+ * Security:
+ * - This is a "write-only" PostHog project API key (not a Personal API Key)
+ * - Cannot be used to read/modify data, only to send events
+ * - PostHog project is configured to reject malicious data
+ *
+ * Privacy:
+ * - All data sent to this instance is covered by Avail's Privacy Policy
+ * - No sensitive data (wallet addresses, amounts) collected by default
+ * - Users can opt-out via analytics.enabled = false
+ *
+ * Override:
+ * Set NEXUS_POSTHOG_KEY env var or pass custom key in SDK config
+ */
+const DEFAULT_POSTHOG_KEY = process.env.NEXUS_POSTHOG_KEY || 'phc_UD6lQU3PEw1d8oo8E17rJLmRAR7kxJbQ5OseHuCvi7N';
+const DEFAULT_POSTHOG_HOST = 'https://us.i.posthog.com';
+
+export class AnalyticsManager {
+  private provider: AnalyticsProvider;
+  private session: SessionManager;
+  private performance: PerformanceTracker;
+  private config: AnalyticsConfig;
+  private enabled: boolean;
+  private sdkVersion: string;
+  private network?: 'mainnet' | 'testnet';
+
+  constructor(config?: AnalyticsConfig, sdkVersion: string = '0.0.1', network?: 'mainnet' | 'testnet') {
+    this.config = config || { enabled: true };
+    this.enabled = config?.enabled !== false; // Default: true (opt-out)
+    this.sdkVersion = sdkVersion;
+    this.network = network;
+
+    // Initialize session and performance trackers
+    this.session = new SessionManager();
+    this.performance = new PerformanceTracker();
+
+    // Initialize the appropriate provider
+    if (this.enabled) {
```

Comment:
Let me know if my thoughts are correct:
The reason why we are not using `this.config.enabled` is because we want to keep this object as it is. That's why we have `this.enabled` so that we can change its value without touching `this.config.enabled`?

Reply:
Yes I checked with AI, `this.enabled` is kept so we can allow devs to toggle this setting on demand dynamically without touching `this.config.enabled`